tools/asset_builder/AssetBuilder.cpp

```diff
@@ -0,0 +1,245 @@
+#include "assimp/Importer.hpp"
+#include "assimp/scene.h"
+#include "assimp/postprocess.h"
+
+#include <cstring>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+/** The AssetBuilder is a utility to convert fbx and obj file to a simple
+ *  format using AssImp. Currently, the only purpose of this is to speed
+ *  up debugging by moving calls to AssImp outside of the engine. */
+
+struct Config
+{
+    std::filesystem::path AssetTargetsPath;
+    std::filesystem::path OutputDirectory;
+};
+
+const auto AssetExtension = std::string{".nca"};
+const auto DefaultAssetTargetFilename = std::string{"targets.txt"};
+constexpr auto AssimpFlags = aiProcess_Triangulate |
+                             aiProcess_JoinIdenticalVertices |
+                             aiProcess_ConvertToLeftHanded |
+                             aiProcess_GenNormals |
+                             aiProcess_CalcTangentSpace;
+
+void Usage();
+bool ParseArgs(int argc, char** argv, Config* config);
+void CreateOutputDirectory(const std::filesystem::path& directory);
+auto ReadTargets(const std::filesystem::path& targetsPath) -> std::vector<std::filesystem::path>;
+bool IsValidMeshExtension(const std::filesystem::path& extension);
+auto ToAssetPath(const std::filesystem::path& meshPath, const Config& config) -> std::filesystem::path;
+void SanitizeFloat(float* value, bool* badValueDetected);
+void SanitizeVector(aiVector3D* value, bool* badValueDetected);
+void BuildAsset(Assimp::Importer* importer, const std::filesystem::path& inPath, const Config& config);
+auto operator<<(std::ostream& stream, aiVector3D& vec) -> std::ostream&;
+
+int main(int argc, char** argv)
+{
+    Config config;
+    if(!ParseArgs(argc, argv, &config))
+    {
+        Usage();
+        return 0;
+    }
+
+    try
+    {
+        CreateOutputDirectory(config.OutputDirectory);
+        Assimp::Importer importer;
+        for(const auto& targetPath : ReadTargets(config.AssetTargetsPath))
+        {
+            BuildAsset(&importer, targetPath, config);
+        }
+    }
+    catch(const std::exception& e)
+    {
+        std::cerr << e.what() << '\n';
+    }
+
+    return 0;
+}
+
+void Usage()
+{
+    std::cout << "Usage: build.exe [options]\n"
+              << "Options:\n"
+              << "  -h or --help            Display this information\n"
+              << "  -f <file>               Read targets from <file>\n"
```

Comment:
I can definitely see myself wanting to specify a single mesh rather than having to write the mesh path to a file first. 
ie. build.exe -m "C:\Temp\MyMesh.fbx" -o "C:\Source\nc\project\assets\mesh"